hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/CompactionHoodiePathCommand.scala

```diff
@@ -0,0 +1,134 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.hudi.command
+
+import org.apache.hudi.{DataSourceUtils, DataSourceWriteOptions, HoodieWriterUtils}
+import org.apache.hudi.client.WriteStatus
+import org.apache.hudi.common.model.HoodieTableType
+import org.apache.hudi.common.table.timeline.{HoodieActiveTimeline, HoodieTimeline}
+import org.apache.hudi.common.table.{HoodieTableMetaClient, TableSchemaResolver}
+import org.apache.hudi.common.util.{Option => HOption}
+import org.apache.hudi.exception.HoodieException
+import org.apache.spark.api.java.{JavaRDD, JavaSparkContext}
+import org.apache.spark.sql.catalyst.expressions.{Attribute, AttributeReference}
+import org.apache.spark.sql.catalyst.plans.logical.CompactionOperation
+import org.apache.spark.sql.{Row, SparkSession}
+import org.apache.spark.sql.catalyst.plans.logical.CompactionOperation.{CompactionOperation, RUN, SCHEDULE}
+import org.apache.spark.sql.execution.command.RunnableCommand
+import org.apache.spark.sql.hudi.HoodieSqlUtils
+import org.apache.spark.sql.types.StringType
+
+import scala.collection.JavaConversions._
+import scala.collection.JavaConverters._
+
+case class CompactionHoodiePathCommand(path: String,
+  operation: CompactionOperation, instantTimestamp: Option[Long] = None)
+  extends RunnableCommand {
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val metaClient = HoodieTableMetaClient.builder().setBasePath(path)
+      .setConf(sparkSession.sessionState.newHadoopConf()).build()
+
+    assert(metaClient.getTableType == HoodieTableType.MERGE_ON_READ,
+      s"Must compaction on a Merge On Read table.")
+    val schemaUtil = new TableSchemaResolver(metaClient)
+    val schemaStr = schemaUtil.getTableAvroSchemaWithoutMetadataFields.toString
+
+    val parameters = HoodieWriterUtils.parametersWithWriteDefaults(
+        HoodieSqlUtils.withSparkConf(sparkSession, Map.empty)(
+          Map(
+            DataSourceWriteOptions.TABLE_TYPE_OPT_KEY.key() -> HoodieTableType.MERGE_ON_READ.name()
+          )
+        )
+      )
+    val jsc = new JavaSparkContext(sparkSession.sparkContext)
+    val client = DataSourceUtils.createHoodieClient(jsc, schemaStr, path,
+      metaClient.getTableConfig.getTableName, parameters)
+
+    operation match {
+      case SCHEDULE =>
```

Comment:
can we just reuse the `CompactionAdminClient` or other tools we use to run compactions. there are classes in org.apache.hudi.utilities.*  for cleaning, compaction, clustering already. more we reuse the better

Reply:
the advantage is all the error checking etc should be pushed into those classes once and we keep the layer here very thin

Reply:
That is good. I will have a try.